processor/filterprocessor/filter_processor_test.go

```diff
@@ -184,15 +185,27 @@ func TestFilterMetricProcessor(t *testing.T) {
 				}
 			}
 
-			cErr := fmp.ConsumeMetricsData(context.Background(), md)
+			cErr := fmp.ConsumeMetrics(
+				context.Background(),
+				pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
+					{},
+					md,
+					{
+						Metrics: []*metricspb.Metric{},
+					},
+				}))
```

Comment:
I might be missing something but I don't quite understand what passing these three elements here achieves.

Also, not necessarily required for this PR, but it seems like it would be simpler if you just changed the above code to generate `pdata.Metrics` and pass this through directly without calling `pdatautil.MetricsFromMetricsData` at all (this test doesn't need to test that function?).

Likewise the below validation code should just validate the returned `pdata.Metrics` rather than converting it. Otherwise this just leaves more stuff that has to be refactored later (given the end goal is to eventually remove `pdatautil.MetricsFromMetricsData` and `pdatautil.MetricsToMetricsData`)

(similar for the other test)

Reply:
Can this test be refactored so that the first and third structs in the input be passed in separate test cases? The fact that the same inputs are being tested for in multiple test cases seem redundant.

Reply:
Chiming in since I wrote the original test.

@asuresh4 could you clarify what you mean? Do you mean the overlap between TestFilterMetricProcessor and BenchmarkFilter_MetricNames? That one was because the Benchmark* tests only run when "-bench" is passed to go test.

Reply:
No. Sorry I wasn't clear. I was referring to this part where 3 elements are passed in each test case: 

```
pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
	{},
	md,
	{
		Metrics: []*metricspb.Metric{},
	},
})
```

Seems like it maybe possible to parametrize the test case to take `pdata.Metrics` (instead of metrics names as input) and then have separate test cases to test the cases that represent the first and third elements?

Reply:
Ah yeah, ok, thanks for clarifying. I agree with you and @james-bebbington the first and third struct here are more edge cases for pdatautil package than the filterprocessor.

Doesn't look like the pdatautil can be avoided yet though for creating the pdata.Metrics:
https://github.com/open-telemetry/opentelemetry-collector/blob/master/consumer/pdata/metric.go#L49

Reply:
Ok, copied the test in a separate test case and duplicate the code to make sure we test for empty and nil metrics.